Album/ViewControllers/DetailsViewController.m

```diff
@@ -6,34 +6,133 @@
 //
 
 #import "DetailsViewController.h"
+#import "AlbumConstants.h"
+#import "UserPin.h"
 #import <Parse/PFImageView.h>
+#import "PhotoCollectionCell.h"
 @interface DetailsViewController ()
 @property (weak, nonatomic) IBOutlet UILabel *placeNameLabel;
 @property (weak, nonatomic) IBOutlet UILabel *dateLabel;
-@property (weak, nonatomic) IBOutlet PFImageView *pinImageView;
 @property (weak, nonatomic) IBOutlet UITextView *captionTextView;
-
+@property (weak, nonatomic) IBOutlet UIButton *likeButton;
+@property (weak, nonatomic) IBOutlet UIPageControl *pageControl;
+@property (nonatomic) int currentIndex;
+@property (strong, nonatomic) PFUser *currentUser;
+@property (strong, nonatomic) UserPin *likeStatus;
 @end
 
 @implementation DetailsViewController
 
 - (void)viewDidLoad {
-	[super viewDidLoad];
-	// Do any additional setup after loading the view.
-	// Set location
-	self.placeNameLabel.text = self.placeName;
-	// Set date
-	self.dateLabel.text = self.date;
-	// Set caption
-	self.captionTextView.text = self.caption;
-	// Set image
-	[self.pinImageView setFile:self.pinImage];
-	[self.pinImage getDataInBackgroundWithBlock:^(NSData *imageData, NSError *error) {
-	         if (!error) {
-			 UIImage *image = [UIImage imageWithData:imageData];
-			 [self.pinImageView setImage:image];
-		 }
-	 }];
+    [super viewDidLoad];
+    // Set current user
+    self.currentUser = [PFUser currentUser];
+    // Set location
+    self.placeNameLabel.text = self.pin.placeName;
+    // Set date
+    // Set the date formatter
+    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+    [formatter setDateFormat:@"MMM dd, YYYY"];
+    [formatter setDateStyle:NSDateFormatterMediumStyle];
+    NSString *date = [formatter stringFromDate:self.pin.traveledOn];
+    self.dateLabel.text = date;
+    // Set caption
+    self.captionTextView.text = self.pin.captionText;
+    // Photo carousel
+    self.imageCarouselView.delegate = self;
+    self.imageCarouselView.dataSource = self;
+    // Set up like status
+    [self setLikeStatus];
+    // Set up page control
+    self.currentIndex = 0;
+    self.pageControl.numberOfPages = 0;
+    // Set button
+    UITapGestureRecognizer *tapGesture = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(tapLiked:)];
+    tapGesture.numberOfTapsRequired = 2;
+    [self.likeButton addGestureRecognizer:tapGesture];
```

Comment:
`viewDidLoad` and `viewWillAppear` methods in my opinion should be as readable and lean as possible, hence you can move some of the set up you're doing here to another methods within this class, maybe one for setting up date, another one for setting up outlets, etc as you see fit

Reply:
Added helper method!